mapping.json

```diff
@@ -1,38 +1,48 @@
-        ["google-analytics.com/inpage_linkid.js", "inpage_linkid.js"],
-        ["google-analytics.com/plugins/ga/inpage_linkid.js", "inpage_linkid.js"],
-        ["google-analytics.com/cx/api.js", "api.js"],
-        ["google-analytics.com/gtm/js", "gtm.js"]
+        { "url": "google-analytics.com/ga.js", "surrogate": "ga.js" },
+        { "url": "google-analytics.com/analytics.js", "surrogate": "analytics.js" },
+        { "url": "google-analytics.com/inpage_linkid.js", "surrogate": "inpage_linkid.js" },
+        { "url": "google-analytics.com/plugins/ga/inpage_linkid.js", "surrogate": "inpage_linkid.js" },
+        { "url": "google-analytics.com/cx/api.js", "surrogate": "api.js" },
+        { "url": "google-analytics.com/gtm/js", "surrogate": "gtm.js" }
     ],
     "googletagmanager.com": [
-        ["googletagmanager.com/gtm.js", "gtm.js"]
+        { "url": "googletagmanager.com/gtm.js", "surrogate": "gtm.js" }
     ],
     "googletagservices.com": [
-        ["googletagservices.com/gpt.js", "gpt.js"],
-        ["googletagservices.com/tag/js/gpt.js", "gpt.js"]
-    ],        
+        { "url": "googletagservices.com/gpt.js", "surrogate": "gpt.js" },
+        { "url": "googletagservices.com/tag/js/gpt.js", "surrogate": "gpt.js" }
+    ],
     "googlesyndication.com": [
-        ["googlesyndication.com/adsbygoogle.js", "adsbygoogle.js"],
-        ["googlesyndication.com/pagead/js/adsbygoogle.js", "adsbygoogle.js"]
+        { "url": "googlesyndication.com/adsbygoogle.js", "surrogate": "adsbygoogle.js" },
+        { "url": "googlesyndication.com/pagead/js/adsbygoogle.js", "surrogate": "adsbygoogle.js" }
     ],
     "doubleclick.net": [
-        ["doubleclick.net/instream/ad_status.js", "ad_status.js"],
-        ["doubleclick.net/tag/js/gpt.js", "gpt.js"]
+        { "url": "doubleclick.net/instream/ad_status.js", "surrogate": "ad_status.js" },
+        { "url": "doubleclick.net/tag/js/gpt.js", "surrogate": "gpt.js" }
     ],
     "scorecardresearch.com": [
-        ["scorecardresearch.com/beacon.js", "beacon.js"]
+        { "url": "scorecardresearch.com/beacon.js", "surrogate": "beacon.js" }
     ],
     "outbrain.com": [
-        ["outbrain.com/outbrain.js", "outbrain.js"]
+        { "url": "outbrain.com/outbrain.js", "surrogate": "outbrain.js" }
     ],
     "amazon-adsystem.com": [
-        ["amazon-adsystem.com/aax2/amzn_ads.js", "amzn_ads.js"]
+        { "url": "amazon-adsystem.com/aax2/amzn_ads.js", "surrogate": "amzn_ads.js" }
     ],
     "chartbeat.com": [
-        ["chartbeat.com/chartbeat.js", "chartbeat.js"],
-        ["chartbeat.com/js/chartbeat.js", "chartbeat.js"]
+        { "url": "chartbeat.com/chartbeat.js", "surrogate": "chartbeat.js" },
+        { "url": "chartbeat.com/js/chartbeat.js", "surrogate": "chartbeat.js" }
+    ],
+    "facebook.net": [
+        { "regexUrl": "facebook\\.net\\/[a-z_A-Z]+\\/sdk\\.js", "surrogate": "fb-sdk.js", "action": "block-ctl-fb" },
```

Comment:
aren't those going to match against notfacebook.com? I guess the domain check that happens first will prevent it?

Reply:
I think we'd have that problem across the board?  Since the other URLs above are also turned into the same style regex.

Reply:
hm, I think we do an initial check against "facebook.net" so I think we should be good here (cc @jdorweiler )